src/foundry/client/apps/forms/journal-page-sheet.d.ts

```diff
@@ -0,0 +1,292 @@
+  }
+  namespace JournalPageSheet {
+    // eslint-disable-next-line @typescript-eslint/no-empty-interface
+    interface Options extends DocumentSheetOptions {}
+    interface SheetData<T extends JournalPageSheet.Options | undefined = JournalPageSheet.Options | undefined> {
+      cssClass: string;
+      editable: boolean;
+      document: JournalEntryPage;
+      data: ReturnType<JournalEntryPage["toObject"]>;
+      limited: JournalEntryPage["limited"];
+      options?: Partial<T>;
+      owner: JournalEntryPage["isOwner"];
+      title: string;
+    }
+  }
+  /**
+   * The Application responsible for displaying and editing a single {@link JournalEntryPage} text document.
+   */
+  class JournalTextPageSheet<
+    TOptions extends JournalTextPageSheet.Options = JournalTextPageSheet.Options
+  > extends JournalPageSheet<TOptions> {
+    /**
+     * Bi-directional HTML \<-\> Markdown converter.
+     */
+    static get _converter(): Showdown.Converter;
+    /**
+     * Declare the format that we edit text content in for this sheet so we can perform conversions as necessary.
+     */
+    static get format(): JOURNAL_ENTRY_PAGE_FORMATS;
+    /**
+     * @defaultValue
+     * ```typescript
+     * const options = super.defaultOptions;
+     * options.classes.push("text");
+     * options.secrets.push({parentSelector: "section"});
+     * ```
+     */
+    static get defaultOptions(): JournalTextPageSheet.Options;
+
+    /**
+     * The table of contents for this JournalTextPageSheet.
+     */
+    toc: ReturnType<typeof JournalEntryPage["buildTOC"]>;
+
+    // FIXME: this should be async once the parent classes are updated
+    override getData(options?: Partial<TOptions> | undefined): JournalPageSheet.SheetData<TOptions>;
+
+    // FIXME: this should be async once the parent classes are updated
+    close(options?: FormApplication.CloseOptions | undefined): Promise<void>;
+
+    // FIXME: this should be async once the parent classes are updated
+    protected _render(
+      force?: boolean | undefined,
+      options?: Application.RenderOptions<TOptions> | undefined
+    ): Promise<void>;
+
+    /**
+     * Determine if any editors are dirty.
+     */
+    isEditorDirty(): boolean;
+
+    // FIXME: this should be async once the parent classes are updated
+    activateEditor(
+      name: string,
+      options?: TextEditor.Options | undefined,
+      initialContent?: string | undefined
+    ): Promise<Editor | EditorView>;
+    // FIXME: this should be async once the parent classes are updated
+    protected _renderInner(data: object): Promise<JQuery<HTMLElement>>;
+    _getSecretContent(secret: HTMLElement): string;
+    // FIXME: revisit this when it's added to its ancestor classes
+    _updateSecret(secret: HTMLElement, content: string): Promise<typeof this>;
+    // FIXME: this should be async once the parent classes are updated
+    protected _updateObject(event: Event, formData: object): Promise<unknown>;
```

Comment:
Nit; another newline between methods would help readability and conform to general style better. Please update this everywhere there isn't an empty line between functions.